### PR TITLE
Next gen General purpose sku recommendation integration changes

### PR DESCRIPTION
**Pull request by stuti149**
This PR has changes to integrate Next Gen General Purpose SKU recommendation with nuget.

SKU recommendation for Next Gen General Purpose is controlled by a flag.

### CODE REVIEW

File: src/Microsoft.SqlTools.Migration/MigrationService.cs [hunk 1]
```diff
@@ -1088,43 +1088,72 @@ internal static List<AzureSqlSkuCategory> GetEligibleSkuCategories(string target
...
                     eligibleSkuCategories.Add(new AzureSqlSkuPaaSCategory(
                                                     AzureSqlTargetPlatform.AzureSqlManagedInstance,
                                                     AzureSqlPurchasingModel.vCore,
                                                     AzureSqlPaaSServiceTier.BusinessCritical,
                                                     ComputeTier.Provisioned,
                                                     AzureSqlPaaSHardwareType.Gen5));
+                    if (recommendNextGen)
+                    {
+                        eligibleSkuCategories.Add(new AzureSqlSkuPaaSCategory(
+                                                        AzureSqlTargetPlatform.AzureSqlManagedInstance,
```
Comment (sahuroshan) [inline src/Microsoft.SqlTools.Migration/MigrationService.cs:1101]: nit : we can combine 3 if statements into single if ?

Comment (stuti149) [reply src/Microsoft.SqlTools.Migration/MigrationService.cs:1101]: There is an order to be followed for categories. To maintain the order it is added differently.

File: bin/nuget/Microsoft.SqlServer.SqlTargetProvisioning.1.0.20240321.1.nupkg
Comment (Ramudaykumar) [inline bin/nuget/Microsoft.SqlServer.SqlTargetProvisioning.1.0.20240321.1.nupkg:1]: I cannot see old NuGet for Target Provisioning getting deleted. please delete that as well.

Comment (stuti149) [reply bin/nuget/Microsoft.SqlServer.SqlTargetProvisioning.1.0.20240321.1.nupkg:1]: It's deleted 
If you see the changes  318 is replaced to 321. Assessment has already two nugets, one is deleted , other is replaced.

image

**Review by sahuroshan — APPROVED** [review]

**Review by YahyaMohamedYahya — COMMENTED** [review]
Thankyou

**Pull request merged**